### ISSUE TITLE
Exception in thread "main" java.lang.NullPointerException

### ISSUE THREAD

**Issue by henriquesandri** [Kotlin/kotlinx-kover#228]
- `Kover 0.6.0`
- `Kotlin 1.7.10`
- `Gradle 7.5.1`

When I run some task to generate xml report, I have this error:

```
Exception in thread "main" java.lang.NullPointerException
	at com.intellij.rt.coverage.report.XMLCoverageReport.writeLine(XMLCoverageReport.java:221)
	at com.intellij.rt.coverage.report.XMLCoverageReport.writeFile(XMLCoverageReport.java:141)
	at com.intellij.rt.coverage.report.XMLCoverageReport.writePackage(XMLCoverageReport.java:123)
	at com.intellij.rt.coverage.report.XMLCoverageReport.writeProject(XMLCoverageReport.java:93)
	at com.intellij.rt.coverage.report.XMLCoverageReport.write(XMLCoverageReport.java:58)
	at com.intellij.rt.coverage.report.Reporter.createXMLReport(Reporter.java:42)
	at com.intellij.rt.coverage.report.Main.main(Main.java:38)
```

I've already tried `koverXmlReport`, `koverMergedXmlReport` and `koverMergedReport`. Html's report is working fine.

Unfortunately my project isnt open source, so I can't share the code.

Thanks for the help.

**Comment by henriquesandri** [comment]
Hi everyone,

@k3marek, thanks for your answer, after I added a few files into `excludes`, my `xmlReport` task is working. I also believe this issue could be solved to improve the Kover, so I also want to kindly ask authors to take a look on this issue.

I would like to thank you in advance for assistance and say that I am available for questions if necessary.

**Comment by shanshin** [comment]
Hi,
@k3marek, thanks for the research, it will help with solving the problem!
@henriquesandri, of course, first of all it should be solved by introducing filters for build types. 
However, the NPE error message does not help users in understand the causes of the error, so the agent also needs to be fixed - a small producer would help with this.

**Comment by shanshin** [comment]
Relates #18

**Comment by henriquesandri** [comment]
@shanshin ,

Sorry for my beginner question, but can I debug the problematic task if I clone the Kover project? If yes, there are any documentation which explain it?

I'd like to obtain more information and maybe propose some solution.

**Comment by tjerkw** [comment]
Same problem for mu multiplatform module

Tried to exclude tasks but didnt work:
```
kover {
    instrumentation {
        excludeTasks += setOf("jsTest", "iosTest") 
    }
}
```

```
> Task :shared:koverXmlReport FAILED
Exception in thread "main" java.lang.NullPointerException
        at com.intellij.rt.coverage.report.XMLCoverageReport.writeLine(XMLCoverageReport.java:221)
        at com.intellij.rt.coverage.report.XMLCoverageReport.writeFile(XMLCoverageReport.java:141)
        at com.intellij.rt.coverage.report.XMLCoverageReport.writePackage(XMLCoverageReport.java:123)
        at com.intellij.rt.coverage.report.XMLCoverageReport.writeProject(XMLCoverageReport.java:93)
        at com.intellij.rt.coverage.report.XMLCoverageReport.write(XMLCoverageReport.java:58)
        at com.intellij.rt.coverage.report.Reporter.createXMLReport(Reporter.java:42)
        at com.intellij.rt.coverage.report.Main.main(Main.java:38)
``

**Comment by shanshin** [comment]
@tjerkw , could you provide a reproducer?
Kover currently only works with JVM tasks, so excluding `jsTest` and `iosTest` won't help in this case.

**Comment by tjerkw** [comment]
I found it:

in `commonMain`
```
expect class PlatformLogOutput() : LogOutput
```

And in `jvmMain` I have an actual.


I solved it now by just ignoring this class:

```
kover {
    filters {
        classes {
            // Kover doesnt work well with actuals:
            // https://github.com/Kotlin/kotlinx-kover/issues/228
            excludes += listOf("my.project.PlatformLogOutput")
        }
    }
}
```

**Comment by shanshin** [comment]
@tjerkw, thanks!

**Comment by zuevmaxim** [comment]
NPE is fixed in agent version `1.0.688`

**Comment by shanshin** [comment]
To use custom engine you need to configure it in all projects where Kover is used

for Kotlin Gradle script
```
kover {
    engine.set(kotlinx.kover.api.IntellijEngine("1.0.688"))
}
```

for Groovy
```
kover {
    engine = kotlinx.kover.api.IntellijEngine("1.0.688")
}
```

**Comment by shanshin** [comment]
Fixed in `0.7.0-Alpha`
